tests/Cookie/SessionCookieJarTest.php

```diff
@@ -67,9 +67,19 @@ public function testPersistsToSession()
         // Load the cookieJar from the file
         $jar = new SessionCookieJar($this->sessionVar);
 
-        // Weeds out temporary and session cookies
-        $this->assertEquals(2, count($jar));
+        if($testSaveSessionCookie) {
+            $this->assertEquals(3, count($jar));
+        } else {
+            // Weeds out temporary and session cookies
+            $this->assertEquals(2, count($jar));
+        }
+
         unset($jar);
         unset($_SESSION[$this->sessionVar]);
     }
+
+    public function testPersistsToSessionWithSessionCookie()
+    {
+        $this->testPersistsToSession(true);
```

Comment:
This should be removed and instead tested using a data provider. See https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.dataProvider